### ISSUE TITLE
Code system references without codes defined throws an exception

### ISSUE THREAD

**Issue by EvanMachusak** [FirelyTeam/firely-cql-sdk#513]
Depends on #508 

```
library Test version '1.0.0'

codesystem "v3.ServiceDeliveryLocationRoleType": 'http://terminology.hl7.org/ValueSet/v3-ServiceDeliveryLocationRoleType'

define f: "v3.ServiceDeliveryLocationRoleType" is CodeSystem
```

Throws a `KeyNotFound` exception during compilation.

Expected value of `f` should be `true` according to the Java cqf ruler.

**Comment by baseTwo** [comment]
In our cql to elm compiler?

**Comment by EvanMachusak** [comment]
Yes.  More crucially, I wrote this test because this will fail.  We create `codesystem` declarations in our classes as `CqlCode[]` because I think you can use `in` on code systems as well as value sets, but this creates type confusion that will cause this operator to return false, and also prevents `ChangeType` from working, because our `TypeResolver` says a `CodeSystemRef` should be a `CqlCodeSystem` primitive.

I think we should change CqlCodeSystem to implement `IEnumerable<Code>` and stop declaring code systems as `CqlCode[]`.
`

**Comment by baseTwo** [comment]
Blocked. Waiting on 
* #519

**Comment by baseTwo** [comment]
Re-verified against `develop` @ `3810aa572`: **the repro in this issue still fails at compile time**, but the symptom has changed, so here is an updated account.

I compiled the exact library from the issue body through `CqlToolkit`/`ElmToolkit`:

```cql
library Test513 version '1.0.0'
codesystem "v3.ServiceDeliveryLocationRoleType": 'http://terminology.hl7.org/ValueSet/v3-ServiceDeliveryLocationRoleType'
define f: "v3.ServiceDeliveryLocationRoleType" is CodeSystem
```

Result:

```
Hl7.Cql.Exceptions.CqlException`1[Hl7.Cql.Compiler.ExpressionBuildingError]:
  Cannot convert Hl7.Cql.Primitives.CqlCode[] to Hl7.Cql.Primitives.CqlCodeSystem.
   at CodeBuilderContext.ChangeType (Cql/Cql.Compiler/CodeBuilderContext.Operators.cs:858)
   at CodeBuilderContext.TranslateElement -> ConvertToResultType (Cql/Cql.Compiler/CodeBuilderContext.cs:223)
   at CodeBuilderContext.Is (Cql/Cql.Compiler/CodeBuilderContext.Operators.cs:464)
```

So:

- **No longer a `KeyNotFound`.** A code system declared without any codes is handled — `Cql/Cql.Compiler/CodeBuilderContext.LibraryDefs.cs:29-30` falls back to an empty list:
  ```csharp
  if (!_libraryContext.TryGetCodesByCodeSystemName(codeSystem.name, out var codes))
      codes = [];
  ```
  That part of the report is fixed.
- **The type confusion diagnosed in the comment thread is still live, but inverted.** Code systems are no longer *emitted* as `CqlCode[]` — `CqlCodeSystemDefinition.ReturnType` is `typeof(CqlCodeSystem)` (`Cql/Cql.Compiler/CqlDefinition.cs:81-88`) and the generated member is a real `CqlCodeSystem` (see `Cql/CoreTests/CSharp/TestCodeSystemInclude-1.0.0.g.cs:28-33`). What was *not* updated is the reference side: `CodeSystemRef` still types the call as an array of codes —
  ```csharp
  // Cql/Cql.Compiler/CodeBuilderContext.Refs.cs:103-111
  var type = _typeResolver.CodeType.MakeArrayType();     // => CqlCode[]
  var definitionCallExpression = InvokeDefinitionThroughRuntimeContext(codeSystemRef.name, codeSystemRef.libraryName, type);
  ```
  while `BaseTypeResolver.CodeSystemType` is `typeof(CqlCodeSystem)` (`Cql/Cql.Runtime/Runtime/BaseTypeResolver.cs:81`), which is what the ELM's `{urn:hl7-org:elm-types:r1}CodeSystem` result type resolves to. `ConvertToResultType` then tries `CqlCode[] -> CqlCodeSystem` and there is no such conversion.
- No test covers a `CodeSystemRef` in expression position: the only `CodeSystemRef` nodes in any checked-in ELM (`Cql/CoreTests/Input/ELM/Test/QueriesTest-1.0.0.json`) sit inside `codeDef.codeSystem`, which goes through `TryGetCodeSystemName` and never touches `CodeBuilderContext.Refs.cs:103`. That is why the regression went unnoticed.

The expected value is still `true` per the spec — `spec/cql/condensed/09-b-cqlreference.md:130` defines the `CodeSystem` type, and `Is` over it is well-formed.

**Still correctly blocked on #519.** Fixing only `CodeBuilderContext.Refs.cs:108` to hand back `CqlCodeSystem` would make this define compile, but the sibling case `"some code" in "some codesystem"` fails with the identical error and additionally has **no operator to bind to** — `Cql/Cql.Runtime/Operators/ICqlOperators.cs` contains zero `CodeSystem` members (only the `*InValueSet` family), while `spec/cql/condensed/09-b-cqlreference.md:6176-6212` specifies six `in (Codesystem)` overloads. That is #519's scope, and #519 is still open.

Keeping this as the concrete regression case to close out when #519 lands. Suggested acceptance: this exact library evaluates `f` to `true`, with the CQL added to the golden/ELM test inputs.

Confidence: high — reproduced directly against a build of the current tree; every cited symbol was opened and checked.
